routing/endpointregistry.go

```diff
@@ -55,24 +89,23 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 			for _, epi := range route.LBEndpoints {
 				metrics := r.GetMetrics(epi.Host)
 				if metrics.DetectedTime().IsZero() {
-					r.SetDetectedTime(epi.Host, now)
+					metrics.SetDetected(now)
```

Comment:
I think we need to also SetDetected in the fadeIn postprocessor that handles endpointCreated filter
https://github.com/zalando/skipper/blob/5f706e5058720a70ecd3427ebe99306a4c918bc9/filters/fadein/fadein.go#L228-L244

Reply:
I don't think so. The responsibility of fadeIn is to compare current time and time endpoint being detected and decide whether the current endpoint will be the candidate for the current request or not.

As long as we have [endpointregistry](https://github.com/zalando/skipper/blob/v0.18.75/skipper.go#L1908) postprocessor being executed before [fadein](https://github.com/zalando/skipper/blob/v0.18.75/skipper.go#L1911) postprocessor the write responsibility is on endpointregistry, the read responsibility is on fadein (and any other components if we would like to have them in the future).

Reply:
yes, but there is https://github.com/zalando/skipper/blob/master/docs/reference/filters.md#endpointcreated that allows override of the detected time

Reply:
Okay, I agree.

Reply:
I was thinking endpointregistry should not care about this filter before, that's why I need to slightly change one more test:
https://github.com/zalando/skipper/pull/2795/files#diff-fe3e40c1b9322e1df1457d8d81bcac816fe66474c1c9d2638a3328a6d70a1e09R434